### PR TITLE
ensure that only one context menu can be opened

### LINKED ISSUE #7635: investigate why typescript integration tests timesout sometimes

### PR DESCRIPTION
**Pull request by akosyakov**
#### What it does

- fix #7635: tests are failing because programmatically without any user input one could open many context menus. This PR makes sure that current context menu is closed before a new is opened.

#### How to test

- CI should be green
- test context menus manually

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: examples/api-tests/src/menus.spec.js [hunk 1]
```diff
@@ -97,4 +108,63 @@ describe('Menus', function () {
...
+
+        const access = contextMenuService.render({
+            anchor: { x: 0, y: 0 },
+            menuPath: contextMenuPath
+        });
+        toTearDown.push(access);
+        if (!(access instanceof BrowserContextMenuAccess)) {
+            assert.isTrue(access instanceof BrowserContextMenuAccess);
+            return;
+        }
```
Comment (kittaakos) [inline examples/api-tests/src/menus.spec.js:132]: I must be blind, but I do not understand this.

File: packages/core/src/browser/context-menu-renderer.ts [hunk 2]
```diff
@@ -16,22 +16,73 @@
...
+                this._current = undefined;
+            }));
+            this.toDisposeOnSetCurrent.push(current);
+        }
+    }
+
     render(options: RenderContextMenuOptions): ContextMenuAccess;
     /** @deprecated since 0.7.2 pass `RenderContextMenuOptions` instead */
     render(menuPath: MenuPath, anchor: Anchor, onHide?: () => void): ContextMenuAccess;
+    render(arg: MenuPath | RenderContextMenuOptions, arg2?: Anchor, arg3?: () => void): ContextMenuAccess {
```
Comment (kittaakos) [inline packages/core/src/browser/context-menu-renderer.ts:77]: `render(menuPath: MenuPath | RenderContextMenuOptions, anchor?: Anchor, onHide?: () => void): ContextMenuAccess`

Comment (akosyakov) [reply packages/core/src/browser/context-menu-renderer.ts:77]: yeah, except the first argument

File: examples/api-tests/src/menus.spec.js (hunk 1 above)
Comment (akosyakov) [reply examples/api-tests/src/menus.spec.js:132]: that's for type checking, ideally i will just do `assert.isTrue(access instanceof BrowserContextMenuAccess)`, but then ts does not proper type of `access`

It should be possible with https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions But our TS versions is below 3.7 :(

Someone should do https://github.com/eclipse-theia/theia/issues/7693

File: packages/core/src/browser/context-menu-renderer.ts (hunk 2 above)
Comment (kittaakos) [reply packages/core/src/browser/context-menu-renderer.ts:77]: I see. You're right, please pick a better name than `arg`.

Comment (akosyakov) [reply packages/core/src/browser/context-menu-renderer.ts:77]: `pathOrOptions`?

Comment (akosyakov) [reply packages/core/src/browser/context-menu-renderer.ts]: went with `menuPathOrOptions`

**Pull request merged**